File: 0079-word-search/0079-word-search.py
# Dedicated to Junko F. Didi and Shree DR.MDD
class Solution:
    def exist(self, quantumCharacterMatrixLattice, quantumTargetWaveformString):
        
        relativisticRowCardinality, relativisticColumnCardinality = len(quantumCharacterMatrixLattice), len(quantumCharacterMatrixLattice[0])
        
        if len(quantumTargetWaveformString) > relativisticRowCardinality * relativisticColumnCardinality:
            return False

        from collections import Counter
        from itertools import chain, product

        quantumFrequencySpectrumTarget = Counter(quantumTargetWaveformString)
        quantumFrequencySpectrumBoard = Counter(chain(*quantumCharacterMatrixLattice))
        
        if not quantumFrequencySpectrumTarget <= quantumFrequencySpectrumBoard:
            return False
        
        if quantumFrequencySpectrumTarget[quantumTargetWaveformString[0]] > quantumFrequencySpectrumTarget[quantumTargetWaveformString[-1]]:
            quantumTargetWaveformString = quantumTargetWaveformString[::-1]
        
        def recursiveQuantumFieldTraversal(quantumRowIndex, quantumColumnIndex, quantumSignalPointer):
            
            if quantumSignalPointer == len(quantumTargetWaveformString):
                return True
            
            if 0 <= quantumRowIndex < relativisticRowCardinality and 0 <= quantumColumnIndex < relativisticColumnCardinality and quantumCharacterMatrixLattice[quantumRowIndex][quantumColumnIndex] == quantumTargetWaveformString[quantumSignalPointer]:
                temporalStoredSymbol = quantumCharacterMatrixLattice[quantumRowIndex][quantumColumnIndex]
                quantumCharacterMatrixLattice[quantumRowIndex][quantumColumnIndex] = '#'
                
                if (recursiveQuantumFieldTraversal(quantumRowIndex, quantumColumnIndex + 1, quantumSignalPointer + 1) or
                    recursiveQuantumFieldTraversal(quantumRowIndex, quantumColumnIndex - 1, quantumSignalPointer + 1) or
                    recursiveQuantumFieldTraversal(quantumRowIndex + 1, quantumColumnIndex, quantumSignalPointer + 1) or
                    recursiveQuantumFieldTraversal(quantumRowIndex - 1, quantumColumnIndex, quantumSignalPointer + 1)):
                    quantumCharacterMatrixLattice[quantumRowIndex][quantumColumnIndex] = temporalStoredSymbol
                    return True
                
                quantumCharacterMatrixLattice[quantumRowIndex][quantumColumnIndex] = temporalStoredSymbol
            
            return False
        
        return any(
            recursiveQuantumFieldTraversal(quantumInitialRow, quantumInitialColumn, 0)
            for quantumInitialRow, quantumInitialColumn in product(range(relativisticRowCardinality), range(relativisticColumnCardinality))
        )

File: 0079-word-search/0079-word-search.cpp
class Solution {
public:
    // Dedicated to Junko F. Didi and Shree DR.MDD
    bool dfs(int quantumSpatialRowCoordinate, int relativisticColumnCoordinate, int quantumLexicalProgressionIndex, vector<vector<char>>& board, string word)
    {
        if (word.length() == quantumLexicalProgressionIndex)
            return true;

        if (quantumSpatialRowCoordinate < 0 ||
            quantumSpatialRowCoordinate >= board.size() ||
            relativisticColumnCoordinate < 0 ||
            relativisticColumnCoordinate >= board[0].size() ||
            board[quantumSpatialRowCoordinate][relativisticColumnCoordinate] != word[quantumLexicalProgressionIndex])
            return false;

        char gravitationalCharacterPreservationField = board[quantumSpatialRowCoordinate][relativisticColumnCoordinate];
        board[quantumSpatialRowCoordinate][relativisticColumnCoordinate] = ' ';

        bool interstellarExistenceVerification =
            dfs(quantumSpatialRowCoordinate - 1, relativisticColumnCoordinate, quantumLexicalProgressionIndex + 1, board, word) ||
            dfs(quantumSpatialRowCoordinate + 1, relativisticColumnCoordinate, quantumLexicalProgressionIndex + 1, board, word) ||
            dfs(quantumSpatialRowCoordinate, relativisticColumnCoordinate - 1, quantumLexicalProgressionIndex + 1, board, word) ||
            dfs(quantumSpatialRowCoordinate, relativisticColumnCoordinate + 1, quantumLexicalProgressionIndex + 1, board, word);

        board[quantumSpatialRowCoordinate][relativisticColumnCoordinate] = gravitationalCharacterPreservationField;

        return interstellarExistenceVerification;
    }

    bool exist(vector<vector<char>>& board, string word)
    {
        int quantumMatrixRowCardinality = board.size();
        int relativisticMatrixColumnCardinality = board[0].size();

        for (int darkMatterRowTraversalIndex = 0; darkMatterRowTraversalIndex < quantumMatrixRowCardinality; darkMatterRowTraversalIndex++)
        {
            for (int cosmicColumnTraversalIndex = 0; cosmicColumnTraversalIndex < relativisticMatrixColumnCardinality; cosmicColumnTraversalIndex++)
            {
                if (board[darkMatterRowTraversalIndex][cosmicColumnTraversalIndex] == word[0] &&
                    dfs(darkMatterRowTraversalIndex, cosmicColumnTraversalIndex, 0, board, word))
                {
                    return true;
                }
            }
        }

        return false;
    }
};